enhancements/ingress/aws-lb-eip-selection.md

```diff
@@ -0,0 +1,474 @@
+
+Obvious valid scenario.
+
+I think we need some feedback from installer-team, @patrickdillon or @mtulio or @sadasu on this type of validation. Should the installer team consider
+the Predicted LB Subnets != BYO Subnet Count scenario as not valid? And possibly block future installs as a resolution
+to https://issues.redhat.com/browse/OCPBUGS-17432? That would make EIP Allocation a lot easier, but not sure if that's realistic. 
+
+4. #### Validation to check if EIPs exist and are not already assigned to resources.
+EIPs can be assigned to many resource types, like Nat Gateways, *LBs, Instances, etc. The attribute associationId will be set when the EIP is already associated.
+To mitigate this we could add that validation, at least on installer, to provide quick-feedback (fail when validate install-config) to the user when the provided EIP is already associated to another resource.
+It would be nice to have a validation before setting the annotation to CCM, keeping the operator degraded before disrupting the service.
+
+5. #### IngressConfig EIP Allocation Defaulting Mechanics for Ingress Controller
+Traditionally, the Ingress Operator has populated default values from the Ingress Config into the `status`, 
+making `status` effectively reflect the desired state of the IngressController. However, since 
+`eipAllocations` in `status` represents the **actual** state, not the **desired** state, the default 
+`eipAllocations` values must be set in the `spec` when the Ingress Operator initially admits the 
+IngressController.
+This approach is new. The Ingress Operator does not typically set default values in `spec` for load balancer
+configurations if the user hasn’t explicitly provided them. While this defaulting pattern is more consistent
+with Kubernetes conventions for `spec` and `status` (and is also our only option in this situation), it's
+important to acknowledge that this inconsistency in defaulting behavior could cause confusion for users.
+
+6. #### Validation to check if lbType was to NLB when eipAllocations were provided in the installer
+EIPs can provided only for `NLB` type `IngressController` so, the installer will be check for the lbType set to NLB when `eipAllocations` are provided
+in the `install-config.yaml`. We can't add a CEL in the Platform API type of the installer so a validation will need to be
+added in the installer code.
+
+### Risks and Mitigations
+
+- Providing a right number of EIPs same as the number of public subnets as per AWS guidelines is an admin
+  responsibility.
+- Also providing existing available EIPs which are not assigned to any instance is important.
+- If the above two points are not followed, the Kubernetes LoadBalancer service for the IngressController
+  won't get assigned EIPs, which will cause the LoadBalancer service to be in persistently pending state by the
+  Cloud Controller Manager (CCM). The reason for the persistently pending state is posted to the status of the IngressController.
```

Comment:
You mean Degraded state, right?

Reply:
The Degradation happens after sometime. Initially we see Pending state.